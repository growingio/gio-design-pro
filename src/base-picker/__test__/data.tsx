import React from 'react';
import { Button } from '@gio-design/components';
import { PlusCircleFilled } from '@gio-design/icons';
import List from '../../list';

export const renderItems = () => (
  <>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
    <List.Item>内容</List.Item>
  </>
);

export const renderGroups = () => (
  <>
    <List.ItemGroup title="分组 1">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
    <List.ItemGroup title="分组 2">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
    <List.ItemGroup title="分组 3">
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
      <List.Item>内容</List.Item>
    </List.ItemGroup>
  </>
);

export const footer = (
  <Button type="text" icon={<PlusCircleFilled />} size="small">
    新建内容
  </Button>
);

export const tabNavItems = [
  { key: 'tab-1', children: '选项 1' },
  { key: 'tab-2', children: '选项 2' },
];
