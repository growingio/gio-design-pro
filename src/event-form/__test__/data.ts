// eslint-disable-next-line import/prefer-default-export
export const spaceTags: any[] = [
  {
    id: 'JnG40OGz',
    name: '是',
    creator: 'haozhigang',
    creatorId: 'rRGoYQml',
    createdAt: '2021-01-14T07:08:25Z',
    updater: 'haozhigang',
    updaterId: 'rRGoYQml',
    updatedAt: '2021-01-14T07:08:25Z',
    description: '是  得到的',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx123456',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx123456',
      path: 'pages/index/index',
      query: '得到的=s',
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'zZDb6aD9',
    name: '1月13，21次',
    creator: 'lvyuqiang',
    creatorId: 'n1QVaDyR',
    createdAt: '2021-01-13T09:03:46Z',
    updater: 'lvyuqiang',
    updaterId: 'n1QVaDyR',
    updatedAt: '2021-01-13T09:03:46Z',
    description: '1月13，21次',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx123456',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx123456',
      path: 'pages/index/index',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'mgGJy2pA',
    name: 'wx123 的是页面 pages/form/form 。',
    creator: 'lvyuqiang',
    creatorId: 'n1QVaDyR',
    createdAt: '2021-01-13T07:14:25Z',
    updater: 'lvyuqiang',
    updaterId: 'n1QVaDyR',
    updatedAt: '2021-01-13T07:14:25Z',
    description: '现在定义的是页面 pages/form/form 。',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/form/form',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx123456',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx123456',
      path: 'pages/form/form',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'AVpZWyGK',
    name: 'ls-xl',
    creator: 'liushuo',
    creatorId: 'qgQMnp3L',
    createdAt: '2021-01-12T05:14:29Z',
    updater: 'liushuo',
    updaterId: 'qgQMnp3L',
    updatedAt: '2021-01-12T05:14:29Z',
    description: '',
    platforms: ['web'],
    docType: 'elem',
    actions: ['chng'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: '/push/cdp/testgio-cdpcircel.html',
      pg: null,
      query: null,
      xpath: '/div.main/div.title/form.basic-grey/label/select.selection',
      contentType: null,
      domain: 'release-messages.growingio.cn',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'release-messages.growingio.cn',
      path: '/push/cdp/testgio-cdpcircel.html',
      query: null,
      xpath: '/div.title/form.basic-grey/label/select.selection',
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/07b18645-b784-4abe-b973-a08605bf5893.jpg?sign=wcyjkYl8LHTacSHpdojrXHeTrHPzHONADaFNg5OBY3U%3D&expires=1611037476177',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
  {
    id: 'WkDyn8Qe',
    name: 'ls-jk',
    creator: 'liushuo',
    creatorId: 'qgQMnp3L',
    createdAt: '2021-01-12T05:13:53Z',
    updater: 'liushuo',
    updaterId: 'qgQMnp3L',
    updatedAt: '2021-01-12T05:13:53Z',
    description: '',
    platforms: ['web'],
    docType: 'elem',
    actions: ['clck'],
    patternMatched: false,
    attrs: {
      content: '查找 query 柯基图片',
      index: null,
      href:
        'https://www.google.com/search?q=%E6%9F%AF%E5%9F%BA%E5%9B%BE%E7%89%87&oq=%E6%9F%AF%E5%9F%BA%E5%9B%BE%E7%89%87&aqs=chrome..69i57j0l2j69i61j69i60j0.4959j0j4&sourceid=chrome&ie=UTF-8',
      path: '/push/cdp/testgio-cdpcircel.html',
      pg: null,
      query: null,
      xpath: '/div.main/div.title/form.basic-grey/a',
      contentType: null,
      domain: 'release-messages.growingio.cn',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'release-messages.growingio.cn',
      path: '/push/cdp/testgio-cdpcircel.html',
      query: null,
      xpath: '/div.main/div.title/form.basic-grey/a',
      index: null,
      href: null,
      content: '查找 query 柯基图片',
      pg: null,
      contentType: '=',
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/364c9f7b-9a27-47c5-94d2-bfe193e3c89a.jpg?sign=bFYchAI2ahfGZykwMOZL3Pm6NBYfc2srQPXu6jt8cC8%3D&expires=1611037476177',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
  {
    id: 'kqQe1qQr',
    name: '定义一个新的事件2',
    creator: 'zhangxueqi',
    creatorId: 'oNGzxQgB',
    createdAt: '2021-01-08T06:21:14Z',
    updater: 'zhangxueqi',
    updaterId: 'oNGzxQgB',
    updatedAt: '2021-01-08T06:21:14Z',
    description: '',
    platforms: ['minp'],
    docType: 'elem',
    actions: ['clck'],
    patternMatched: false,
    attrs: {
      content: '主页',
      index: '0',
      href: 'pages/index/index',
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: '#onTabItemTap',
      contentType: null,
      domain: 'wx265d0fa6fa70fae9',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: '12=12',
      xpath: '#onTabItemTap',
      index: '0',
      href: 'pages/index/index',
      content: '主页',
      pg: null,
      contentType: '=',
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: '7RDYqYDA',
    name: '事件2的页面',
    creator: 'zhangxueqi',
    creatorId: 'oNGzxQgB',
    createdAt: '2021-01-08T06:20:52Z',
    updater: 'zhangxueqi',
    updaterId: 'oNGzxQgB',
    updatedAt: '2021-01-08T06:20:52Z',
    description: '',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx265d0fa6fa70fae9',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: '12=12',
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'ExQWkgQ7',
    name: '事件4',
    creator: 'zhangxueqi',
    creatorId: 'oNGzxQgB',
    createdAt: '2021-01-08T06:19:40Z',
    updater: 'zhangxueqi',
    updaterId: 'oNGzxQgB',
    updatedAt: '2021-01-08T06:19:40Z',
    description: '',
    platforms: ['minp'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: 'pages/index/index',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'wx265d0fa6fa70fae9',
      urlScheme: null,
      __typename: 'DocProps',
    },
    definition: {
      domain: 'wx265d0fa6fa70fae9',
      path: 'pages/index/index',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: null,
      __typename: 'DocProps',
    },
    screenshot: { target: '', viewport: '', __typename: 'Screenshot' },
    __typename: 'Element',
  },
  {
    id: 'xNQXjGMb',
    name: 'autotracker-cdp页面',
    creator: 'mayutong',
    creatorId: 'nxGK0Da6',
    createdAt: '2020-12-15T07:50:07Z',
    updater: 'mayutong',
    updaterId: 'nxGK0Da6',
    updatedAt: '2020-12-15T07:50:07Z',
    description: '',
    platforms: ['Android'],
    docType: 'page',
    actions: ['page'],
    patternMatched: false,
    attrs: {
      content: null,
      index: null,
      href: null,
      path: '/TrackActivity',
      pg: null,
      query: null,
      xpath: null,
      contentType: null,
      domain: 'com.growingio.android.cdp',
      urlScheme: 'growing.447120b8e608d6b9',
      __typename: 'DocProps',
    },
    definition: {
      domain: 'com.growingio.android.cdp',
      path: '/TrackActivity',
      query: null,
      xpath: null,
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: null,
      urlScheme: 'growing.447120b8e608d6b9',
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/2740844b-cb23-4868-8a43-d0f5cd150cc8.jpg?sign=LV5Clhvs36wRzmvx83nGR9J76ZvqDFp%2F1D54C7PixOs%3D&expires=1611037476178',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
  {
    id: 'nxGKXxDa',
    name: '验收限定条件2',
    creator: '马雨彤',
    creatorId: 'evp9bjGO',
    createdAt: '2021-02-08T03:34:14Z',
    updater: '马雨彤',
    updaterId: 'evp9bjGO',
    updatedAt: '2021-02-08T03:34:14Z',
    description: '不勾选',
    platforms: ['Android'],
    docType: 'elem',
    actions: ['clck'],
    patternMatched: false,
    attrs: {
      content: '加载了 并且初始化了 SDK',
      index: null,
      href: null,
      path: '/push/cdp/testgio-cdpcircel.html',
      pg: null,
      query: null,
      xpath: '/div.title/div/div.title/h3',
      contentType: null,
      domain: 'release-messages.growingio.cn',
      urlScheme: 'growing.314b5db864634231',
      __typename: 'DocProps',
    },
    definition: {
      domain: 'release-messages.growingio.cn',
      path: '/push/cdp/testgio-cdpcircel.html',
      query: null,
      xpath: '/div.title/div/div.title/h3',
      index: null,
      href: null,
      content: null,
      pg: null,
      contentType: '=',
      urlScheme: 'growing.314b5db864634231',
      __typename: 'DocProps',
    },
    screenshot: {
      target: '',
      viewport:
        '/screenshots-uploads/screenshots/ecfb4779-46d1-4a10-84b0-1818a9d721d8.jpg?sign=%2FCv2NaAELe7kyLCW7WRrJLckE%2F8X7QL%2FLx5A3FzdG6Y%3D&expires=1612852820607',
      __typename: 'Screenshot',
    },
    __typename: 'Element',
  },
];

export const deviceInfoMinp = { name: '小程序无埋点', domain: 'wx123456', os: 'minp' };
export const currentPageMinp = {
  id: 'f339fcb2bdd07fd7b55020c2835d8bede823e4af',
  eventType: 'PAGE',
  timestamp: 1610954652616,
  ip: '111.198.136.131',
  domain: 'wx123456',
  query: 'name=hello&a=3',
  platform: 'MinP',
  path: 'pages/index/index',
  appType: 0,
  actions: ['clck'],
};
export const deviceInfoApp = {
  projectId: '91eaf9b283361032',
  msgType: 'ready',
  timestamp: 1612755238635,
  domain: 'com.growingio.android.cdp',
  sdkVersion: '3.0.0',
  sdkVersionCode: 30000,
  os: 'Android',
  screenWidth: 1080,
  screenHeight: 1920,
  urlScheme: 'growing.314b5db864634231',
  name: 'Android无埋点测试',
};
export const currentPageApp = {
  path: '/MainActivity',
  title: 'AutoTracker-CDP',
  left: 0,
  top: 0,
  width: 343.125,
  height: 610,
  appType: 1,
  domain: 'com.growingio.android.cdp',
  isDefined: false,
};
