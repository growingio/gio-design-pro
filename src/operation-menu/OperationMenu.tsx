/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import Dropdown from '@gio-design/components/es/components/dropdown';
import Button from '@gio-design/components/es/components/button';
import { MoreOutlined } from '@gio-design/icons';
import { OperationMenuProps } from './interfaces';
import OperationMenuList from './OperationMenuList';

const OperateMenu = (props: OperationMenuProps) => {
  const {
    options,
    width = 160,
    trigger = ['click'],
    mini = true,
    size,
    icon = <MoreOutlined />,
    placement = 'bottomLeft',
    buttonType = 'assist',
    iconClassName,
  } = props;

  return (
    <div onClick={(e: any) => e.stopPropagation()}>
      <Dropdown
        overlay={
          <OperationMenuList width={width} options={options} onClick={(option: any) => props.onClick?.(option)} />
        }
        placement={placement}
        trigger={trigger}
        onVisibleChange={(visible: boolean) => props.onVisibleChange?.(visible)}
      >
        <Button
          className={iconClassName}
          mini={mini}
          size={size}
          onClick={(e: any) => e.stopPropagation()}
          icon={icon}
          type={buttonType}
        />
      </Dropdown>
    </div>
  );
};

export default OperateMenu;
