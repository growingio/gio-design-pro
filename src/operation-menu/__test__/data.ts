export const options = [
  {
    disabled: false,
    tooltip: '',
    value: 'update-space-role',
    label: '修改站点角色',
  },
  {
    disabled: false,
    tooltip: '',
    value: 'update-space',
    label: '删除用户',
  },
  {
    disabled: true,
    tooltip: '行行行',
    value: 'update-spxxxace',
    label: '删除用户2',
  },
];

export default {
  options,
};
