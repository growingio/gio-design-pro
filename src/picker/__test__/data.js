export const nodes = [
  {
    label: 'Data 1',
    value: 'data-1',
  },
  {
    label: 'Data 2',
    value: 'data-2',
  },
  {
    label: 'Data 3',
    value: 'data-3',
  },
];

export default {
  nodes,
};
