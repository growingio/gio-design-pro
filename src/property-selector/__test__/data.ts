// eslint-disable-next-line import/prefer-default-export
export const insightDimensions = [
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'd',
    name: '域名',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'user',
    groupName: '用户属性',
    id: 'usr_t_date_1',
    name: '测试日期型',
    type: 'usr',
    valueType: 'date',
  },
  {
    groupId: 'user',
    groupName: '用户属性',
    id: 'usr_t_number_1',
    name: '测试数值型',
    type: 'usr',
    valueType: 'int',
  },
  {
    groupId: 'user',
    groupName: '用户属性',
    id: 'usr_sex002',
    name: 'sex002',
    type: 'usr',
    valueType: 'string',
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'rd',
    name: '访问来源',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'rt',
    name: '一级访问来源',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'kw',
    name: '搜索词',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_source',
    name: '广告来源',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_campaign',
    name: '广告名称',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_content',
    name: '广告内容',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_term',
    name: '广告关键字',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'utm_medium',
    name: '广告媒介',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'origin',
    groupName: '用户来源',
    id: 'ch',
    name: '自定义App渠道',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'city',
    name: '城市',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'region',
    name: '地区',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'countryCode',
    name: '国家代码',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'geo',
    groupName: '地域信息',
    id: 'countryName',
    name: '国家名称',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'bw',
    name: '浏览器',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'bwv',
    name: '浏览器版本',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'os',
    name: '操作系统',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'osv',
    name: '操作系统版本',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'shw',
    name: '屏幕大小（高*宽）',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'l',
    name: '操作系统语言',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'db',
    name: '设备品牌',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'dm',
    name: '设备型号',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'device',
    groupName: '设备信息',
    id: 'o',
    name: '设备方向',
    type: 'global',
    valueType: null,
  },
  {
    id: 'AbQ3bDYe',
    isPrimaryKey: false,
    key: 'order_name',
    type: 'itm',
    groupId: 'item',
    name: 'shangpin',
    groupName: '物品属性',
    valueType: 'string',
    description: '',
    associatedId: 'var_engageType',
  },
  {
    associatedId: '',
    groupId: 'event',
    groupName: '事件变量',
    id: 'var_engageType',
    name: '产品交互性参与类型',
    type: 'event',
    valueType: 'string',
  },
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'tm',
    name: '时间我是超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字超级长的名字',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'b',
    name: '应用平台',
    type: 'global',
    valueType: null,
  },
  {
    groupId: 'normal',
    groupName: '常用维度',
    id: 'cv',
    name: '应用版本',
    type: 'global',
    valueType: null,
  },
];
