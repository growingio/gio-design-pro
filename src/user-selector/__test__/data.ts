export const currentUserId = 'V0G5vDAR';

export const segment = {
  id: 'y9pm1pme',
  name: '过去 7 天浏览过详情页的地活跃用户',
  detector: {
    usersRatio: 0.002,
    totalUsers: 32,
  },
  creator: '李四',
  creatorId: 'V0G5vDAR',
  createdAt: '2020-05-11T03:45:31.275Z',
  updatedAt: '2020-05-11T03:45:31.275Z',
};

export const segments = [
  {
    id: 'y9pm1pme',
    name: '过去 7 天浏览过详情页的地活跃用户',
    detector: {
      usersRatio: 0.002,
      totalUsers: 32,
    },
    creator: '李四',
    creatorId: 'V0G5vDAR',
    createdAt: '2020-05-11T03:45:31.275Z',
    updatedAt: '2020-05-11T03:45:31.275Z',
  },
  {
    id: 'AbQ3bDYe',
    name: '过去 7 天低活跃的免费用户',
    detector: {
      usersRatio: 0.0,
      totalUsers: 0,
    },
    creator: '李四',
    creatorId: 'V0G5vDAR',
    createdAt: '2020-05-11T04:01:53.514Z',
    updatedAt: '2020-05-11T04:01:53.514Z',
  },
  {
    id: 'rRGoYQml',
    name: '过去 7 天访问用户',
    detector: {
      usersRatio: 0.042,
      totalUsers: 832,
    },
    creator: '李四',
    creatorId: 'V0G5vDAR',
    createdAt: '2020-05-20T08:32:30.387Z',
    updatedAt: '2020-05-20T08:32:30.387Z',
  },
  {
    id: '3KlwQik2',
    name: '过去 7 天登录用户',
    detector: {
      usersRatio: 0.014,
      totalUsers: 247,
    },
    creator: '王武',
    creatorId: 'o8D03G7P',
    createdAt: '2020-05-20T08:32:30.387Z',
    updatedAt: '2020-05-20T08:32:30.387Z',
  },
  {
    id: 'WlGk4Daj',
    name: '未转化用户',
    detector: {
      usersRatio: 0.059,
      totalUsers: 129347,
    },
    creator: '张三',
    creatorId: '7RDYeDA8',
    createdAt: '2020-05-09T02:15:20.643Z',
    updatedAt: '2020-05-09T02:15:20.643Z',
  },
  {
    id: 'klGvyp7E',
    name: '已转化用户',
    detector: {
      usersRatio: 0.034,
      totalUsers: 112379,
    },
    creator: '陈留',
    creatorId: 'nKD1PGWr',
    createdAt: '2020-05-22T02:44:06.972Z',
    updatedAt: '2020-05-22T02:44:06.972Z',
  },
];

export default {
  segment,
};
