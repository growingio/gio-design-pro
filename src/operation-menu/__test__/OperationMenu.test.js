import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { options } from './data';
import OperationMenu from '../OperationMenu';

describe('OperationMenu', () => {
  it('test operation-menu', () => {
    const onClickMock = jest.fn();
    const wrapper = render(<OperationMenu options={options} onClick={onClickMock} />);

    expect(wrapper).toMatchSnapshot();
    expect(screen.queryByText('修改站点角色')).toBeNull();

    fireEvent.click(screen.getByRole('button'));
    expect(screen.queryByText('修改站点角色')).toBeTruthy();

    fireEvent.click(screen.getByText('修改站点角色'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
