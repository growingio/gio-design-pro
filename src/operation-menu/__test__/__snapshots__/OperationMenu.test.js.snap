// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationMenu test operation-menu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="gio-btn gio-btn-text gio-btn-mini gio-btn-icon-only"
          style="cursor: pointer;"
          type="button"
        >
          <span
            aria-label="more-outlined"
            class="gio-icon"
            role="img"
          >
            <svg
              fill="currentColor"
              height="16px"
              viewBox="0 0 14 14"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="more-outlined_svg__cls-2"
                d="M7 1a1 1 0 11-1 1 1 1 0 011-1zm0 5a1 1 0 11-1 1 1 1 0 011-1zm0 5a1 1 0 11-1 1 1 1 0 011-1z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="gio-btn gio-btn-text gio-btn-mini gio-btn-icon-only"
        style="cursor: pointer;"
        type="button"
      >
        <span
          aria-label="more-outlined"
          class="gio-icon"
          role="img"
        >
          <svg
            fill="currentColor"
            height="16px"
            viewBox="0 0 14 14"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="more-outlined_svg__cls-2"
              d="M7 1a1 1 0 11-1 1 1 1 0 011-1zm0 5a1 1 0 11-1 1 1 1 0 011-1zm0 5a1 1 0 11-1 1 1 1 0 011-1z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
